### ISSUE TITLE
Enforce ordering of items in `schemas` and `lenses` during `BindLineage`

### ISSUE THREAD

**Issue by sdboyer** [grafana/thema#174]
#173 removed certain native CUE logic that sorted `lenses` and `schemas` in a lineage into a known order. To replace this, we need to add checks to `BindLineage()` that enforce ordering invariants on the raw lineage inputs.

**Comment by sdboyer** [comment]
@undef1nd this is the one you were aiming to pick up, right?

**Comment by undef1nd** [comment]
Yes, I think it's one of those errors handling improvements you suggested.
